yt_dlp/extractor/bilibili.py

```diff
@@ -502,39 +503,146 @@ def _real_extract(self, url):
             season_info.get('bangumi_title'), season_info.get('evaluate'))
 
 
-class BilibiliChannelIE(InfoExtractor):
-    _VALID_URL = r'https?://space.bilibili\.com/(?P<id>\d+)'
-    _API_URL = "https://api.bilibili.com/x/space/arc/search?mid=%s&pn=%d&jsonp=jsonp"
+class BilibiliSpaceBaseIE(InfoExtractor):
+    def __init__(self, downloader=None):
+        InfoExtractor.__init__(self, downloader)
+        self.page_idx_start = 1
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+
+        metadata, pagedlist = self._extract_internal(playlist_id)
+
+        return self.playlist_result(pagedlist, playlist_id, metadata['title'])
```

Comment:
If it's a Base extractor you shouldn't have _real_extract, you can rename it to  _extract_playlist and later call it

Also, usually you pre-declare the functions that you use, so that _real_extract is at the bottom

Reply:
ok

Reply:
imo, defining _real_extract in base class is fine as long as it is used by ALL subclasses. It's overriding concrete methods in subclasses that usually creates a mess

good:
```py
class BaseIE:
  def _real_extract(...): ...

class AIE(BaseIE):
    ...

class BIE(BaseIE):
  def _real_extract(...):
     ...
     super()._real_extract(...)
```

bad:
```py
class BaseIE:
  def _real_extract(...): ...

class AIE(BaseIE):
    ...

class BIE(BaseIE):
  def _real_extract(...):
     ... # No super call
```